compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/IntermediateTextInputUIView.uikit.kt

```diff
@@ -233,7 +237,12 @@ internal class IntermediateTextInputUIView(
             location.toInt() to length.toInt()
         }
         val relativeTextRange = locationRelative until locationRelative + lengthRelative
-        input?.setMarkedText(markedText, relativeTextRange)
+
+        // Due to iOS specifics, [setMarkedText] can be called several times in a row. Batching
+        // helps to avoid text input problems caused by too frequent dispaltching of input commands.
```

Comment:
Sorry for late review.

Could you add an explanation in the comment about the exact issue?

I remember you told that it is related to desync of `value` during recomposition and `onValueChanged` callback, but I don't remember how exactly.

Reply:
Could you also add a sequence of events? I am trying to understand how batching will help with this issue, but can't figure it out yet.

If we explain it in the comment, it will be helpful for all future maintainers.

Before the PR:
```
setMarkedText -> onValueChange -> value = "1"
setMarkedText -> onValueChange -> value = "12"
setMarkedText -> onValueChange -> value = "123"
scene.render()   // recomposition -> recompose TextField("123")
```
After this PR:
```
setMarkedText
setMarkedText
setMarkedText
flush() -> 3x onValueChange (value = "1", value = "12", value = "123")
scene.render()   // recomposition -> recompose TextField("123")
```

`scene.render()` can't be called between synchronous `setMarkedText` and can't interfere. But even if it can, it recomposes the last received value via `onValueChange`, isn't it?

Reply:
> scene.render() can't be called between synchronous setMarkedText and can't interfere. But even if it can, it recomposes the last received value via onValueChange, isn't it?

It can be called in between, but the issue itself occurs when there is no recomposition.

> I am trying to understand how batching will help with this issue, but can't figure it out yet.

The batching joins frequent changes. It fixes scenario with update pattern like A -> B -> A, where A value is remembered inside a Composition and the code does not recognises it as a change.

@igordmn , I updated the comments, however you should know some details about it:
1. The current fix is very narrow, it fixes particular input use-case on a particular platform. It does not eliminate the problem entirely.
2. The problem with holding values in composition looks like a design issue which require more dramatic changed and/or rethinking the way we're handling input events on our platforms. The same issue can be reproduced on Desktop as well.

Reply:
> The batching joins frequent change

Sorry, I still can't connect the dots 🫠. Batching should only help in one case - if there some non-synchronous `commit`, `set` operation in between consequential `setMarkedText`:

```
setMarkedText -> onEditCommand
set state
setMarkedText -> onEditCommand
set state
```

It is only possible with 2 conditions:
- if `set state` is called from some `LaunchedEffect`, independently of `render` and events.
- if multiple `setMarkedText` aren't synchronous, and can allow other events happen in-between multiple calls.

If this is the case, then we potentially wire our new logic (batching) with implementation details in the other place of the code. If these implementation details changed, batching can be broken.

If `updateState` is synchronous:
```
setMarkedText -> onEditCommand -> updateState
setMarkedText -> onEditCommand -> updateState
```
Then I am not sure how batching fixes the bug 🤔.

> however you should know some details about it:

Sure, no need to fix it completely. From my perspective I would want that we aren't introducing new ambiguity (so other maintainers can understand the logic) and other potential bugs.